.circleci/main.yml

```diff
@@ -1574,7 +1574,8 @@ jobs:
             --password $LABEL_STUDIO_PASSWORD \
             --user-token $LABEL_STUDIO_USER_TOKEN
 
-            sleep 20  # Wait for Label Studio server to start
+            # Wait for Label Studio server to start
+            while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:8080/api/projects -H "Authorization: Token $LABEL_STUDIO_USER_TOKEN")" != "200" ]]; do sleep 5; done
```

Comment:
I think polling is definitely better, but we should cap the time here. Something long like 5 minutes or 10 minutes just so circle-ci doesn't wait on this on in-progress changes. in the wrong circumstances we've had the ci run for a long time.

Reply:
Does the job requiring `label-studio-ci-testing-docker-build` (defined in the workflows section) not cover this?

Reply:
The Docker image is created before this `test-label-studio` job. After the image is downloaded and run, it takes a few seconds for the Label Studio server to spin up which is what the polling is waiting for. This way we don't run the Python tests before the Label Studio server is ready to handle requests.